source/common/common/fancy_logger.h

```diff
@@ -62,28 +77,69 @@ class FancyContext {
    */
   FancyLogLevelMap getAllFancyLogLevelsForTest() ABSL_LOCKS_EXCLUDED(fancy_log_lock_);
 
+  /**
+   * Updates the all the loggers based on the verbosity updates <(file, level) ...>.
+   * It supports file basename and glob "*" and "?" pattern, eg. ("foo", 2), ("foo/b*", 3)
+   * Patterns including a slash character are matched against full path names, while those
+   * without are matched against base names (by removing one suffix) only.
+   *
+   * It will store the current verbosity updates and clear all previous modifications for
+   * future check when initializing a new logger.
+   *
+   * Files are matched against globs in updates in order, and the first match determines
+   * the verbosity level.
+   *
+   * Files which do not match any pattern use the value of default log level from Context.
+   */
+  void updateVerbositySetting(const std::vector<std::pair<absl::string_view, int>> updates)
+      ABSL_LOCKS_EXCLUDED(fancy_log_lock_);
+
+  /**
+   * Check if a string matches a glob pattern. It only supports "*" and "?" wildcards,
+   * and wildcards may match /. No support for bracket expressions [...].
+   */
+  static bool safeFileNameMatch(absl::string_view pattern, absl::string_view str);
+
 private:
   /**
    * Initializes sink for the initialization of loggers, needed only in benchmark test.
    */
   void initSink();
 
   /**
-   * Creates a logger given key and log level, and add it to map.
+   * Creates a logger given key, and add it to map. Log level is from getLogLevel.
    * Key is the log component name, e.g. file name now.
    */
-  spdlog::logger* createLogger(std::string key, int level = -1)
+  spdlog::logger* createLogger(std::string key) ABSL_EXCLUSIVE_LOCKS_REQUIRED(fancy_log_lock_);
```

Comment:
while we're at it, make this a const ref string

Reply:
Done